Allow transient ICE disconnection.

bemasc opened the pull request:
This change also reduces the data channel send buffer to zero.

Fixes https://github.com/uProxy/uproxy/issues/1511: although ICE disconnection events still occur, they are transient, and do not kill the connection.

Note that removing the sensitivity to ICE disconnection is safe in part because we have a heartbeat to detect long-term connection failures.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/uproxy/uproxy-lib/174)

<!-- Reviewable:end -->

bemasc committed:
Allow transient ICE disconnection.

This change also reduces the data channel send buffer to zero.

trevj commented:
Nice. Seems like we were being over-zealous in our treatment of disconnection, so it's also great to clear that up.

Just a couple of tiny comments, then :+1:!

---





<sup>**[src/samples/copypaste-socks-chromeapp/freedom-module.ts, line 88 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/174#-Jp7aHZxvtuSJRSsbTey)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/f718d538ffdc743048668d8e04ab2cdaacf19e3e/src/samples/copypaste-socks-chromeapp/freedom-module.ts#L88)):</sup>
Again, do you think now is the time to move to churn? :-)

---

<sup>**[src/webrtc/datachannel.ts, line 25 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/174#-Jp7ab9NMidu0RMORMAJ)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/f718d538ffdc743048668d8e04ab2cdaacf19e3e/src/webrtc/datachannel.ts#L25)):</sup>
zero-indexed bullet points? :-)

---

<sup>**[src/webrtc/datachannel.ts, line 45 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/174#-Jp7cw6h_kIkYKhQFgHf)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/f718d538ffdc743048668d8e04ab2cdaacf19e3e/src/webrtc/datachannel.ts#L45)):</sup>
I was going to suggest we chase the sweet spot here, of buffer size vs. throughput -- but I guess even that won't come close to the improvement a drastic architectural change could make.

---

<sup>**[src/webrtc/peerconnection.ts, line 317 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/174#-Jp7drzCiDYHZBTiHRA_)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/f718d538ffdc743048668d8e04ab2cdaacf19e3e/src/webrtc/peerconnection.ts#L317)):</sup>
I realise now there's no way for clients to see these events, so I filed an issue:
https://github.com/uProxy/uproxy/issues/1519

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/174)
<!-- Sent from Reviewable.io -->

bemasc committed:
Don't enable obfuscation by default just yet.

bemasc committed:
Don't use zero-indexed english.

bemasc commented:
<sup>**[src/samples/copypaste-socks-chromeapp/freedom-module.ts, line 88 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/174#-Jp7aHZxvtuSJRSsbTey)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/f718d538ffdc743048668d8e04ab2cdaacf19e3e/src/samples/copypaste-socks-chromeapp/freedom-module.ts#L88)):</sup>
Reverted for now.  Maybe we should enable this once we have negotiation.

---

<sup>**[src/webrtc/datachannel.ts, line 25 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/174#-Jp7ab9NMidu0RMORMAJ)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/f718d538ffdc743048668d8e04ab2cdaacf19e3e/src/webrtc/datachannel.ts#L25)):</sup>
Changed.

---

<sup>**[src/webrtc/datachannel.ts, line 45 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/174#-Jp7cw6h_kIkYKhQFgHf)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/f718d538ffdc743048668d8e04ab2cdaacf19e3e/src/webrtc/datachannel.ts#L45)):</sup>
We talked in person and concluded that we should land this change and then spend a few minutes tuning this parameter.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/174)
<!-- Sent from Reviewable.io -->
